src/backend/sgx/attestation.rs

```diff
@@ -434,6 +444,73 @@ pub fn get_quote(report: &[u8], akid: Vec<u8>, out_buf: &mut [u8]) -> Result<usi
     Ok(quote.len())
 }
 
+/// Gets quote size with CRL added
+pub fn get_quote_size_with_collateral(akid: Vec<u8>) -> Result<usize, Error> {
+    let q_size = get_quote_size(akid)?;
+    let out_buf_temp = vec![0; q_size];
+
+    let mut crl_file = sgx_cache_dir()
+        .map_err(|e| Error::new(ErrorKind::Other, format!("SGX CRL read error {e}")))?;
+    crl_file.push("crls.der");
+
+    let evidence = SgxEvidence {
+        quote: &out_buf_temp,
+        crl: Document::read_der_file(&crl_file)
+            .context(format!("error reading Intel CRL file {crl_file:?}"))
+            .unwrap(),
+    };
+
+    let evidence_vec = evidence
+        .to_vec()
+        .map_err(|e| Error::new(ErrorKind::Other, format!("SGX evidence to DER error {e}")))?;
+
+    Ok(evidence_vec.len())
+}
+
+pub fn get_quote_and_collateral(
+    report: &[u8],
+    akid: Vec<u8>,
+    out_buf: &mut [u8],
+) -> Result<usize, Error> {
```

Comment:
```suggestion
) -> io::Result<usize> {
```

Reply:
This would make the return type inconsistent with the return type in the other functions in `attestation.rs`.